src/components/Pagination.svelte

```diff
@@ -0,0 +1,124 @@
+<script context="module">
+  export const paginateData = (array, totalPages) => {
+    let result = [];
+    let localData = array.slice();
+    for (let i = totalPages; i > 0; i -= 1) {
+      result.push(localData.splice(0, Math.ceil(localData.length / i)));
+    }
+    return result;
+  };
+
+  export const makePages = (page, perPage, data) => {
+    if (data.length === 0) return [];
+    let total = data.length;
+    let currentPage = page;
+    let lastPage = Math.ceil(total / perPage);
+    let from = page + perPage * (page - 1);
+    let to = page * perPage;
+    let pages = paginateData(data, lastPage);
+    return {
+      total,
+      currentPage,
+      lastPage,
+      from,
+      to,
+      pages,
+    };
+  };
+
+  export const goToPage = (page, pages) => {
+    let from = 0;
+    for (let i = 0; i < page - 1; i += 1) {
+      from += pages[i].length;
+    }
+    let to = from + pages[page - 1].length;
+    return {
+      currentPage: page,
+      from,
+      to,
+      page: pages[page - 1],
+    };
+  };
+</script>
+
+<script>
+  import { createEventDispatcher } from "svelte";
```

Comment:
I don't think we actually need an event dispatcher here. Try just exporting an optional callback function -- see the FilterInput component for an example of this:

https://github.com/mozilla/glean-dictionary/blob/9990cef3e5e04a06aaeb6d69470516325e5e8535/src/components/FilterInput.svelte

Reply:
@wlach replacing the event dispatcher has proved a little complex given the need to pass some parameters around unlike with the example shared.

Don't know if I am missing something, I checked a few self-contained `paginators` and they use this approach as well. Please shed more light

Reply:
To be 100% honest with you, I am somewhat new to Svelte as well. 😄 Ok, let's assume this approach is good for now. We can always circle back and improve the component later.